runtime/doc/dev_arch.txt

```diff
@@ -0,0 +1,51 @@
+*dev_internals.txt*          Nvim
+
+
+                            NVIM REFERENCE MANUAL
+
+
+How to develop Nvim, explaination of modules and subsystems    *dev-arch*
```

Comment:
introducing dev_arch where we can explain the overall system and provide instructions for common dev tasks (such as adding a new UI event)

Reply:
Small typo: s/explaination/explanation/